sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -948,6 +952,210 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("SPARK-48280: Expression Walker for Testing") {
```

Comment:
Can you add new test suite (new class) for expression walker?
Especially given that we plan to add operator and photon support.